txclib/wizard.py

```diff
@@ -0,0 +1,149 @@
+import os
+import inquirer
+from txclib import messages
+from slugify import slugify
+from txclib.api import Api
+from txclib.project import Project
+from txclib.log import logger
+from txclib.parsers import set_parser
+
+
+def validate_source_file(answers, path):
+    return os.path.isfile(os.path.abspath(path))
+
+
+def validate_expression(answers, expression):
+    return '<lang>' in expression
+
+
+class Wizard(object):
+
+    def __init__(self, path_to_tx):
+        p = Project(path_to_tx)
+        self.host = p.config.get('main', 'host')
+        username, token_or_password = p.getset_host_credentials(self.host)
+        self.api = Api(username=username, password=token_or_password,
+                       host=self.host, path_to_tx=p.txrc_file)
+
+    def get_organizations(self):
+        try:
+            organizations = self.api.get('organizations')
+        except Exception as e:
+            logger.error(unicode(e))
+            exit(1)
+        return [(o['name'], o['slug']) for o in organizations]
+
+    def get_projects_for_org(self, organization):
+        try:
+            projects = self.api.get('projects', organization=organization)
+        except Exception as e:
+            logger.error(unicode(e))
+            exit(1)
+        return [p for p in projects if not p['archived']]
+
+    def get_formats(self, filename):
+        _, extension = os.path.splitext(filename)
+        try:
+            formats = self.api.get('formats')
+        except Exception as e:
+            logger.error(unicode(e))
+            exit(1)
+
+        def display_format(v):
+            return '{} - {}'.format(v['description'], v['file-extensions'])
+
+        formats = [(display_format(v), k) for k, v in formats.items()
+                   if extension in v['file-extensions']]
+        return formats
+
+    def run(self):
+        """
+        Runs the interactive wizard for `tx set` command and populates the
+        parser's options with the user input. Options `local` and `execute`
+        are by default True when interactive wizard is run.
+
+        Returns: the populated (options, args) tuple.
+        """
+
+        TEXTS = messages.TEXTS
+        options, args = set_parser().parse_args([])
+        options.local = True
+        options.execute = True
+        logger.info(TEXTS['source_file']['description'])
+
+        ans = inquirer.prompt([
+            inquirer.Text('source_file',
+                          message=TEXTS['source_file']['message'],
+                          validate=validate_source_file)
+        ], raise_keyboard_interrupt=True)
+        options.source_file = ans['source_file']
```

Comment:
Might I be so bold as to make a suggestion? I don't know Inquirer's full capabilities but wouldn't it be awesome if we displayed the contents of `os.listdir(',')` and allow them to traverse through directories and even use a text-filter until the user has found the source file?

Reply:
For the record: 

- https://pypi.python.org/pypi/iterfzf/0.2.0.17.0
- `iterfzf((os.path.join(dirpath, filename)[2:] for dirpath, _, filenames in os.walk('.') for filename in filenames))`

Reply:
This is a very nice suggestion @kbairak! I will check how well it can play with inquirer